server/proxy/install.go

```diff
@@ -122,44 +119,36 @@ func (p *Proxy) ensureOAuthApp(app *apps.App, noUserConsent bool, actingUserID s
 	oauth2CallbackURL := p.conf.GetConfig().AppURL(app.AppID) + config.PathMattermostOAuth2Complete
 
 	// For the POC this should work, but for the final product I would opt for a RPC method to register the App
-	oauthApp, response := asAdmin.CreateOAuthApp(&model.OAuthApp{
+	oauthApp := &model.OAuthApp{
 		CreatorId:    actingUserID,
 		Name:         app.DisplayName,
 		Description:  app.Description,
 		CallbackUrls: []string{oauth2CallbackURL},
 		Homepage:     app.HomepageURL,
 		IsTrusted:    noUserConsent,
-	})
-	if response.StatusCode != http.StatusCreated {
-		if response.Error != nil {
-			return nil, errors.Wrap(response.Error, "failed to create OAuth2 App")
-		}
-		return nil, errors.Errorf("failed to create OAuth2 App: received status code %v", response.StatusCode)
+	}
+	err := client.CreateOAuthApp(oauthApp)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create OAuth2 App")
 	}
 
 	p.mm.Log.Debug("App install flow: Created OAuth2 App", "id", oauthApp.Id)
 
 	return oauthApp, nil
 }
 
-func (p *Proxy) ensureBot(app *apps.App, client *model.Client4) error {
+func (p *Proxy) ensureBot(client MMClient, app *apps.App) error {
 	bot := &model.Bot{
 		Username:    strings.ToLower(string(app.AppID)),
 		DisplayName: app.DisplayName,
 		Description: fmt.Sprintf("Bot account for `%s` App.", app.DisplayName),
 	}
 
-	var fullBot *model.Bot
-	var response *model.Response
-	user, _ := client.GetUserByUsername(bot.Username, "")
+	user, _ := client.GetUserByUsername(bot.Username)
 	if user == nil {
-		fullBot, response = client.CreateBot(bot)
-		if response.StatusCode != http.StatusCreated {
-			if response.Error != nil {
-				return response.Error
-			}
-
-			return errors.New("could not create bot")
+		err := client.CreateBot(bot)
```

Comment:
Is the function objective to create or to ensure the bot?

Reply:
To create. It should not exists, given that https://github.com/mattermost/mattermost-plugin-apps/blob/20597e47b253887e53616fb5fe83ab063b7a6a5c/server/proxy/install.go#L147-L147 returned `nil`.